src/hildebrand/internal/response.cljc

```diff
@@ -163,8 +175,13 @@
 
 (defmethod restructure-response* :hildebrand.response/batch-write-item
   [_ {:keys [unprocessed] :as resp}]
-  (or (maybe-unprocessed-error unprocessed)
-      (with-meta {} resp)))
+  (let [result (with-meta
```

Comment:
This reads like it'll return a map with keys `:put`, `:delete`, in the event that there were unprocessed items, though I believe the previous behaviour was to issue an error with the incorrectly formatted result map as the exception data.  I kind of feel like it makes sense to continue to treat unprocessed items as an error, and just supply it with a correctly formatted map per your changes above - what do you think?